### ISSUE TITLE
[Enhancement] Make collapse title usable in routed modals

### ISSUE THREAD

**Issue by MadsBuchmann** [kirbydesign/designsystem#2192]
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
I would like for it to be possible to use collapsible titles in routed modals. 

## Describe the solution you'd like
We need to properly tech refine this to come up with a good solution. 

The modal configs for routed modals are hard coded and can not be changed by consumers. This also means that collapsible titles as introduced in #2174 can not be used for routed modals as they are enabled by setting `ModaConfig.collapseTitle = true`. 

I currently got two ideas for solving this: either we create a `collapseTitle` input property on the `kirby-page-title` element or find a way to edit the modal config in routed modals. 

If we go with the first solution i think we need to check up on how `kirby-page-title` is used. As far as i can see it is only used in the modal but is named after the title component. 

If it is intended for embedding in other components than the modal - i do not think we should go with adding a `collapseTitle` input property to `kirby-page-title` as it will only work in the case where it is embedded in a modal. 

## Are there any additional context?
It was originally a part of #2174 to solve this. To properly think it through and not create a poor solution it has been seperated into its own issue. 

Also #2174 contains breaking changes that should be part of v6.0.0 - the solution for this issue should hopefully just be an extension and contain no breaking changes. 

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [stable branch](https://github.com/kirbydesign/designsystem/tree/stable) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".
